addons/html_builder/static/src/plugins/image/image_shape_option_plugin.js

```diff
@@ -362,15 +366,32 @@ export class ImageShapeOptionPlugin extends Plugin {
 export class SetImageShapeAction extends BuilderAction {
     static id = "setImageShape";
     static dependencies = ["imageShapeOption"];
-    async load({ editingElement: img, value: shapeId }) {
-        const params = { shape: shapeId };
+    async load({ editingElement: img, value: shapeId, params: { mainParam: className } }) {
+        const params = { shape: shapeId, className: className || "" };
         // todo nby: re-read the old option method `setImgShape` and be sure all the logic is in there
         return this.dependencies.imageShapeOption.loadShape(img, params);
     }
-    apply({ editingElement: img, loadResult: updateImageAttributes }) {
+    apply({
+        editingElement: img,
+        loadResult: updateImageAttributes,
+        isPreviewing,
+        params: { mainParam: className },
+    }) {
         updateImageAttributes();
-        const imgFilename = img.dataset.originalSrc.split("/").pop().split(".")[0];
-        img.dataset.fileName = `${imgFilename}.svg`;
+        if (isPreviewing) {
+            //  Hover preview behavior
+            if (className) {
+                img.classList.add(className);
+            }
+            return;
+        }
+        if (className) {
+            img.classList.add(className);
+        }
+        const imgFilename = img.dataset.originalSrc.split("/").pop()?.split(".")[0];
+        if (imgFilename) {
+            img.dataset.fileName = `${imgFilename}.svg`;
+        }
```

Comment:
You can avoid repeating code here:
```suggestion
        if (className) {
            img.classList.add(className);
        }
        if (!isPreviewing) {
            const imgFilename = img.dataset.originalSrc.split("/").pop()?.split(".")[0];
            if (imgFilename) {
                img.dataset.fileName = `${imgFilename}.svg`;
            }
        }
```